ui/components/app/snaps/snap-permission-cell/snap-permission-cell.js

```diff
@@ -0,0 +1,121 @@
+import PropTypes from 'prop-types';
+import { SubjectType } from '@metamask/permission-controller';
+import React from 'react';
+import {
+  AvatarBase,
+  AvatarBaseSize,
+  AvatarFavicon,
+  AvatarFaviconSize,
+  IconSize,
+  Text,
+} from '../../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  Display,
+  FontWeight,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { getAvatarFallbackLetter } from '../../../../helpers/utils/util';
+import PermissionCell from '../../permission-cell';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+
+export default function SnapPermissionCell({
+  snapId,
+  showOptions,
+  permission,
+  connectionSubjectMetadata,
+  index,
+  revoked,
+  approved,
+}) {
+  const t = useI18nContext();
+  let { label, description, leftIcon } = permission;
+
+  if (permission.connection) {
+    if (connectionSubjectMetadata?.subjectType === SubjectType.Snap) {
+      label = t('snapConnectTo', [
+        <Text
+          key="connectToMain"
+          fontWeight={FontWeight.Medium}
+          variant={TextVariant.inherit}
+          color={TextColor.inherit}
+        >
+          {connectionSubjectMetadata.name}
```

Comment:
We can't really use `connectionSubjectMetadata.name` for Snaps as they may have a localized name. 

That's what we use `getSubjectName` for in `permission.js`

Reply:
Updated.